File: lib/api/public/p_matches.ts
import { Match } from "@/types/matches";
import { API_BASE_URL } from "@/lib/config";

export async function fetchUpcomingMatches(): Promise<Match[]> {
  const response = await fetch(`${API_BASE_URL}/matches/upcoming`);
  if (!response.ok) {
    throw new Error("Failed to fetch upcoming matches");
  }
  return response.json();
}

export async function fetchPreviousMatches(): Promise<Match[]> {
  const response = await fetch(`${API_BASE_URL}/matches/previous`);
  if (!response.ok) {
    throw new Error("Failed to fetch previous matches");
  }
  return response.json();
}

export async function fetchUpcomingMatchesByTeam(team: string): Promise<Match[]> {
  const response = await fetch(`${API_BASE_URL}/matches/team?team=${team}`);
  if (!response.ok) {
    throw new Error("Failed to fetch upcoming matches");
  }
  return response.json();
}

File: newBackend/api/public/p_matches.go
package apiPublic

import (
	"backend/services"
	"encoding/json"
	"net/http"
)

func GetUpcomingMatches(w http.ResponseWriter, r *http.Request) {

	if r.Method != http.MethodGet {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}
	upcomingMatches, err := services.GetUpcomingMatches()
	if err != nil {
		http.Error(w, "Failed to fetch upcoming matches", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(upcomingMatches)
}

func GetPreviousMatches(w http.ResponseWriter, r *http.Request) {

	if r.Method != http.MethodGet {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}
	previousMatches, err := services.GetPreviousMatches()
	if err != nil {
		http.Error(w, "Failed to fetch previous matches", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(previousMatches)
}

func GetMatchesByTeam(w http.ResponseWriter, r *http.Request) {

	if r.Method != http.MethodGet {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}
	team := r.URL.Query().Get("team")
	matches, err := services.GetMatchesByTeam(team)
	if err != nil {
		http.Error(w, "Failed to fetch upcoming matches", http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(matches)
}
